core/msg.c

```diff
@@ -208,16 +208,33 @@ int msg_send_int(msg_t *m, kernel_pid_t target_pid)
 int msg_send_receive(msg_t *m, msg_t *reply, kernel_pid_t target_pid)
 {
     assert(sched_active_pid != target_pid);
+
     unsigned state = irq_disable();
     thread_t *me = (thread_t*) sched_threads[sched_active_pid];
     sched_set_status(me, STATUS_REPLY_BLOCKED);
     me->wait_data = (void*) reply;
 
+#ifdef MODULE_CORE_PRIORITY_INHERITANCE
+    thread_t *recv_thread = (thread_t*) sched_threads[target_pid];
+    uint8_t prio_backup = recv_thread->priority;
+    /* When priority inheritance is enabled, we lend the receiver thread our
+     * current priority if lower */
+    if (me->priority < recv_thread->priority) {
+        sched_change_priority(recv_thread, me->priority);
```

Comment:
Shouldn't interrupts be disabled when calling?

Reply:
They are disabled 11 lines up.

Reply:
Sorry, I was apparently blind.